### PR TITLE
update gh to ado sync workflow to use service principals

### PR DESCRIPTION
**Pull request by willtsai**
As explained in https://github.com/radius-project/radius/issues/7402, we must switch to using Service Principals for auth when running the GH-ADO sync workflow.

Auth using Azure Service Principals was added as a part of v2.3 of the `danhellem/github-actions-issue-to-work-item` GitHub Actions workflow we use. Reference: https://github.com/danhellem/github-actions-issue-to-work-item/pull/143

### CODE REVIEW

File: .github/workflows/devops-boards.yaml [hunk 1]
```diff
@@ -13,15 +13,45 @@ jobs:
...
+      #     ado_token: '${{ secrets.ADO_AOCTO_BOT_TOKEN }}'
+      #     github_token: '${{ secrets.GH_RAD_CI_BOT_PAT }}'
+      #     ado_organization: 'azure-octo'
+      #     ado_project: 'Incubations'
+      #     ado_area_path: "Incubations\\Radius"
+      #     ado_iteration_path: "Incubations\\Radius"
+      #     ado_wit: 'GitHub Issue'
+      #     ado_new_state: 'New'
+      #     ado_active_state: 'Active'
+      #     ado_close_state: 'Closed'
```
Comment (rynowak) [inline .github/workflows/devops-boards.yaml:57]: I don't think there's a need for this. If we need to undo this change, it's just a `git revert` - that's even better.

**Review by willdavsmith — APPROVED** [review]

**Review by rynowak — APPROVED** [review]
